x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -347,98 +268,55 @@ export class MapsAppView extends React.Component {
     const globalFilters = filterManager.getGlobalFilters();
     const allFilters = [...savedQueryFilters, ...globalFilters];
 
-    if (savedQuery.attributes.timefilter) {
-      if (savedQuery.attributes.timefilter.refreshInterval) {
-        this._onRefreshChange({
-          isPaused: savedQuery.attributes.timefilter.refreshInterval.pause,
-          refreshInterval: savedQuery.attributes.timefilter.refreshInterval.value,
-        });
-      }
-      this._onQueryChange({
-        filters: allFilters,
-        query: savedQuery.attributes.query,
-        time: savedQuery.attributes.timefilter,
-      });
-    } else {
-      this._onQueryChange({
-        filters: allFilters,
-        query: savedQuery.attributes.query,
+    const refreshInterval = _.get(savedQuery, 'attributes.timefilter.refreshInterval');
+    if (refreshInterval) {
+      this._onRefreshConfigChange({
+        isPaused: refreshInterval.pause,
+        interval: refreshInterval.value,
       });
     }
+    this._onQueryChange({
+      filters: allFilters,
+      query: savedQuery.attributes.query,
+      time: savedQuery.attributes.timefilter,
+    });
   }
 
-  _syncStoreAndGetFilters() {
-    const {
-      savedMap,
-      setGotoWithCenter,
-      setMapSettings,
-      setIsLayerTOCOpen,
-      setOpenTOCDetails,
-    } = this.props;
-    let savedObjectFilters = [];
-    if (savedMap.mapStateJSON) {
-      const mapState = JSON.parse(savedMap.mapStateJSON);
-      setGotoWithCenter({
+  _initMap() {
+    this._initMapAndLayerSettings();
+
+    this.props.clearUi();
+
+    if (this.props.savedMap.mapStateJSON) {
+      const mapState = JSON.parse(this.props.savedMap.mapStateJSON);
```

Comment:
We're parsing `this.props.savedMap.mapStateJSON` in `_initMap()` and `_initMapAndLayerSettings()`. We could just do it once at the top of `initMap()` and then pass `mapState` through to `_initMapAndLayerSettings()`

Reply:
We're actually parsing it again in `getInitialQuery`, so this should remove consolidate 3 duplicate parses down to 1

Reply:
Agreed this is a worthwhile optimization. I would like to leave for another PR so this PR can just focus on removing items from state.